src/briefcase/commands/base.py

```diff
@@ -465,6 +451,51 @@ def verify_tools(self):
         """
         pass
 
+    def finalize_app_config(self, app: BaseConfig):
+        """Finalize the application config.
+
+        Some app configurations (notably, Linux system packages like .deb) have
+        configurations that are deeper than other platforms, because they need
+        to include components that are dependent on command-line arguments. They
+        may also require the existence of system tools to complete
+        configuration.
+
+        The final app configuration merges those "deep" properties into the app
+        configuration, and performs any other app-speceific platform
+        configuration and verification that is required as a result of
+        command-line arguments.
+
+        :param app: The app configuration to finalize.
+        """
+        pass
+
+    def finalize(self, app: Optional[BaseConfig] = None):
```

Comment:
I wanted to sync up on this use of `Optional`.  Personally, I prefer to omit `Optional` when the default value for an argument is `None`.

python/peps#689 updated PEP 484 to stop requiring type checkers to implicitly assume an `Optional` type when the parameter's default is `None`. You can then see mypy's reaction in python/mypy#9091.

This seems to imply it is necessary to manually annotate all parameters with a default value of `None` with `Optional`. However, I don't think that is the intended state of affairs.

In the [definition](https://docs.python.org/3/library/typing.html#typing.Optional) of `Optional`, it says:

> Note that this is not the same concept as an optional argument, which is one that has a default. An optional argument with a default does not require the Optional qualifier on its type annotation just because it is optional. For example:
> 
> ```
> def foo(arg: int = 0) -> None:
>     ...
> ```
> On the other hand, if an explicit value of None is allowed, the use of Optional is appropriate, whether the argument is optional or not. For example:
> 
> ```
> def foo(arg: Optional[int] = None) -> None:
>     ...
> ```

Therefore, `Optional` is really a signal to type checkers (and presumably humans) that calls to the function can explicitly set the argument to `None`. Without `Optional`, setting the argument to `None` is a typing error.

So, I take away from this that unless there's a good reason to "encourage" callers to explicitly pass `None` in some situation, I do not type the parameter with `Optional`.

Thoughts?

Reply:
To call it out explicitly, I think a noticeable hole in my assessment is multiple parameters in a signature that default to `None` when the signature doesn't limit calls to keyword-only.

Therefore, `foo(a: int = None, b: int = None)` could be invoked with `foo(None, 3)` but without `Optional`, a type checker will get upset about the first argument being `None`.

IMO, positional arguments aren't that great and should normally be avoided. Therefore, I'd want the type error.

Alternatively, I think simply adding `Optional` in this case would be fine, too. I just mainly think that most parameters with the `None` default probably shouldn't be typed `Optional`.

Reply:
Actually, I think this is moot :/ It came back to me that the last project I retroactively typed ended up requiring `Optional` everywhere.

Example:
```
✦ ❯ cat test.py 
def foo(a: str = None):
    print(a)

foo(a=None)

✦ ❯ mypy test.py 
test.py:1: error: Incompatible default for argument "a" (default has type "None", argument has type "str")  [assignment]
test.py:1: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
test.py:1: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
test.py:5: error: Argument "a" to "foo" has incompatible type "None"; expected "str"  [arg-type]
Found 2 errors in 1 file (checked 1 source file)
```

Using `--implicit-optional` avoids the error for the signature and the call.
```
✦ ❯ mypy test.py --implicit-optional 
Success: no issues found in 1 source file
```

Of course, using `Optional` also fixes this.
```
✦ ❯ cat test.py 
from typing import Optional

def foo(a: Optional[str] = None):
    print(a)

foo(a=None)

✦ ❯ mypy test.py 
Success: no issues found in 1 source file
```

The back and forth in python/typing#275 definitely shows a desire to still assume `Optional` at least for parameters defaulting to `None`. I wish they had also gone through with renaming `Optional`....alas

Reply:
My understanding is that `Optional[X]` effectively reduces to `X | None`; it's just a more obvious (if more verbose) way of writing it.

If I'm following your discussion here, I think you've come to the conclusion that nothing is needed here... have I understood that correctly?

Reply:
Yeah; nothing needed here. I just needed to talk it out with myself :)